packages/app/src/runner/index.ts

```diff
@@ -37,6 +37,12 @@ export function createWebsocket (socketIoRoute: string) {
 
   const ws = client(socketConfig)
 
+  ws.on('connect_error', () => {
+    // fall back to polling if websocket fails to connect (webkit)
+    // https://github.com/socketio/socket.io/discussions/3998#discussioncomment-972316
+    ws.io.opts.transports = ['polling', 'websocket']
```

Comment:
I always assumed socket.io did this automatically (fallback to polling). I guess not - good to know.

Reply:
Yeah, I thought so too. According to that Gh discussion, it only falls back if the websocket API is completely missing, it doesn't fall back on network errors. In WebKit, the API exists, but the connection itself is failing for some reason.